lib/puppet/datatypes/impl/error.rb

```diff
@@ -1,39 +1,37 @@
 class Puppet::DataTypes::Error
-  attr_reader :message, :kind, :issue_code, :partial_result, :details
+  attr_reader :msg, :kind, :issue_code, :details
+  alias message msg
```

Comment:
Is this intended to support access via Ruby, or Puppet, or both?

Reply:
This function is only accessible via Ruby. If we want it in Puppet, i.e. `$err.message` then we need a function in the Data Type description.

Reply:
I decided to leave it as ruby only for now. It's easier to add it to the datatype than remove it.

Reply:
I think we'll want to add it, because `message` seems more familiar to me than `msg` when working in a manifest. However, I understand that the [task specification](https://github.com/puppetlabs/puppet-specifications/tree/master/tasks#exitcode) uses `msg`. We have an inconsistency that seems best served by providing both accessors, which you can use in different contexts.